core/storage/user/gae_models_test.py

```diff
@@ -2218,3 +2218,98 @@ def test_create_raises_error_when_many_id_collisions_occur(self):
 
         with assert_raises_regexp_context_manager, get_by_id_swap:
             user_models.PseudonymizedUserModel.get_new_id('exploration')
+
+
+class UserAuthModelTests(test_utils.GenericTestBase):
+    """Tests for UserAuthModel."""
+
+    NONEXISTENT_USER_ID = 'id_x'
+    NONREGISTERED_GAE_ID = 'gae_id_x'
+    USER_1_ID = 'user_1_id'
+    USER_1_GAE_ID = ['gae_1_id']
+    USER_2_ID = 'user_2_id'
+    USER_2_GAE_IDs = ['gae_2_id_1', 'gae_2_id_2']
+
+    def setUp(self):
+        """Set up user models in datastore for use in testing."""
+        super(UserAuthModelTests, self).setUp()
+
+        user_models.UserAuthModel(
+            id=self.USER_1_ID,
+            gae_id=self.USER_1_GAE_ID,
+        ).put()
+        user_models.UserAuthModel(
+            id=self.USER_2_ID,
+            gae_id=self.USER_2_GAE_IDs,
+        ).put()
+
+    def test_get_export_policy(self):
+        self.assertEqual(
+            user_models.UserAuthModel.get_export_policy(),
+            base_models.EXPORT_POLICY.NOT_APPLICABLE)
+
+    def test_get_deletion_policy(self):
+        self.assertEqual(
+            user_models.UserAuthModel.get_deletion_policy(),
+            base_models.DELETION_POLICY.DELETE)
+
+    def test_apply_deletion_policy(self):
+        user_models.UserAuthModel.apply_deletion_policy(
+            self.USER_1_ID)
+        self.assertIsNone(
+            user_models.UserAuthModel.get_by_id(
+                self.USER_1_ID
+            )
+        )
+
+        user_models.UserAuthModel.apply_deletion_policy(
+            self.USER_2_ID)
+        self.assertIsNone(
+            user_models.UserAuthModel.get_by_id(
+                self.USER_2_ID
+            )
+        )
+
+        # Test that calling apply_deletion_policy with no existing model
+        # doesn't fail.
+        user_models.UserAuthModel.apply_deletion_policy(
+            self.NONEXISTENT_USER_ID)
+
+    def test_has_reference_to_user_id(self):
+        self.assertTrue(
+            user_models.UserAuthModel
+            .has_reference_to_user_id(self.USER_1_ID)
+        )
+        self.assertTrue(
+            user_models.UserAuthModel
+            .has_reference_to_user_id(self.USER_2_ID)
+        )
+        self.assertFalse(
+            user_models.UserAuthModel
+            .has_reference_to_user_id(self.NONEXISTENT_USER_ID)
+        )
+
+    def test_get_by_auth_id_for_unregistered_auth_id(self):
+        self.assertIsNone(
+            user_models.UserAuthModel
+            .get_by_auth_id('gae', self.NONREGISTERED_GAE_ID)
+        )
+
+    def test_get_by_auth_id_for_single_element_in_auth_id_attribute(self):
+        self.assertEqual(
+            user_models.UserAuthModel.get_by_id(self.USER_1_ID),
+            user_models.UserAuthModel
+            .get_by_auth_id('gae', self.USER_1_GAE_ID[0])
+        )
+
+    def test_get_by_auth_id_for_list_in_auth_id_attribute(self):
```

Comment:
Add a test for a case where get_by_auth_id is given an invalid auth id (i.e. `get_by_auth_id('invalid_auth_id', self.USER_2_GAE_IDS[0]`)

Reply:
Done. Added a test method named `test_get_by_auth_id_with_invalid_auth_type_name` to test for invalid auth type name.